src/js/components/footer/footer.js

```diff
@@ -0,0 +1,93 @@
+// NOTE: This is a template Component1
+
+// Define dependencies on top of the module
+import $ from 'jquery';
+import 'jquery-ui';
+import Base from 'components/base/base.js';
+import Factory from 'helpers/Factory.js';
+
+// Then define Component's defaults
+// Use uppercase for all keys and sub-keys
+const DEFAULTS = {
+    SELECTORS: {
+        // NOTE: Use only js-hooks to elements
+        HIDE_LIST: '.js-footer-toogle',
+        CHILD_LIST: '.js-footer-list'
+    },
+    MODIFIERS: {
+        HIDE_LIST: 'js-footer-list-is-hiden'
+    },
+    NAME_SPACE: '.footer'
+};
+
+// NOTE: export need for tests
+// WARN: Always extend Base component to provide shared API (see Base class)
+export class Footer extends Base {
+    /**
+     * Constructor
+     * @param node {element} - root element for Component's scope
+     */
+    constructor(node) {
+        super(node);
+
+        const $root = $(node);
+        const options = $root.data('options') || {};
+
+        // NOTE: Port to provide extensions/options
+        this.options = Object.assign({}, DEFAULTS, options.Footer || {});
+
+        // Define needed jQuery objects
+        // NOTE: Be consistent and use the context (ex $root.find) to not outcome from the scope
+        // Here might be exceptions as window or document
+        // NOTE: You'll rarely need exact child jQuery objects
+        this.elements = {
+            $root
+        };
+        // If we need some actions on initialization call it in constructor
+        this.hideList();
+    }
+
+    /**
+     * Attaches event listeners
+     */
+    attachEvents() {
+
+    }
+
+    /**
+     * Hide list element from footer
+     */
+    hideList() {
+        this.elements.$root
```

Comment:
As I can see this listener should execute only on mobile/tablet. You can use CSS property pointer-events to prevent the reflow via class toggling.